lib/workload/stateful/filemanager/deploy/lib/filemanager_stack.ts

```diff
@@ -0,0 +1,126 @@
+import { RemovalPolicy, Stack, StackProps } from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as lambdaDestinations from 'aws-cdk-lib/aws-lambda-destinations';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import { IngestFunction, IngestFunctionSettings } from '../constructs/functions/ingest';
+import { Database, DatabaseSettings } from '../constructs/database';
+import { SubnetType } from 'aws-cdk-lib/aws-ec2';
+import { SqsDestination } from 'aws-cdk-lib/aws-s3-notifications';
+import { Bucket, EventType } from 'aws-cdk-lib/aws-s3';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { Queue } from 'aws-cdk-lib/aws-sqs';
+import { Secret } from 'aws-cdk-lib/aws-secretsmanager';
+import { CdkResourceInvoke } from '../constructs/cdk_resource_invoke';
+import { MigrateFunction } from '../constructs/functions/migrate';
+import * as fn from '../constructs/functions/function';
+
+/**
+ * Common settings for the filemanager stack.
+ */
+type Settings = DatabaseSettings &
+  IngestFunctionSettings & {
+    /**
+     * The name of the database. Defaults to `filemanager`.
+     */
+    databaseName?: string;
+    /**
+     * Whether to initialize a database migration.
+     */
+    migrateDatabase?: boolean;
+  };
+
+/**
+ * Stack used to deploy filemanager.
+ */
+export class FilemanagerStack extends Stack {
+  constructor(scope: Construct, id: string, props: StackProps, settings?: Settings) {
+    super(scope, id, props);
+
+    const queue = new Queue(this, 'Queue');
+
+    const testBucket = new Bucket(this, 'Bucket', {
+      bucketName: 'filemanager-test-ingest',
+      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
+      encryption: s3.BucketEncryption.S3_MANAGED,
+      enforceSSL: true,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+    const testBucketPolicy = new PolicyStatement({
+      actions: ['s3:List*', 's3:Get*'],
+      resources: ['arn:aws:s3:::*'],
+    });
+    testBucket.addEventNotification(EventType.OBJECT_CREATED, new SqsDestination(queue));
+    testBucket.addEventNotification(EventType.OBJECT_REMOVED, new SqsDestination(queue));
+
```

Comment:
Not for this PR - but what will be the general story about subscribing to buckets.. presumably the file manger is not going to be responsible for creating all the BYOB buckets (I mean they probably exist completely separate to orcabus).

Reply:
Yes this will need to be different in the future. I'm not sure where the best spot to put this is though. I guess from the filemanager's perspective it just needs access to the SQS queue.

Reply:
https://github.com/aws/aws-cdk/issues/2004

Reply:
So @andrewpatto, if I understand that issue correctly, you'd re-define the CDK stack manually each time a new bucket is introduced?

I was thinking that perhaps getting rid altogether of S3 bucket notification definitions in CDK and do it more dynamically after the stack is deployed? Otherwise we'll have to re-deploy at each new bucket that filemanager needs to watch, not sure if that'll be practical 🤔